deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/SimpleAgentJanitor.java

```diff
@@ -43,30 +42,25 @@
     private static final Logger LOG = LoggerFactory.getLogger(SimpleAgentJanitor.class);
     private AgentDAO agentDAO;
     protected HostDAO hostDAO;
-    protected HostAgentDAO hostAgentDAO;
     private GroupDAO groupDAO;
+    private HostHandler hostHandler;
     protected long maxStaleHostThreshold;
     protected long minStaleHostThreshold;
 
     public SimpleAgentJanitor(ServiceContext serviceContext, int minStaleHostThreshold,
         int maxStaleHostThreshold) {
         agentDAO = serviceContext.getAgentDAO();
         hostDAO = serviceContext.getHostDAO();
-        hostAgentDAO = serviceContext.getHostAgentDAO();
         groupDAO = serviceContext.getGroupDAO();
+        hostHandler = new HostHandler(serviceContext);
         this.maxStaleHostThreshold = maxStaleHostThreshold * 1000;
         this.minStaleHostThreshold = minStaleHostThreshold * 1000;
     }
 
     // remove the stale host from db
     void removeStaleHost(String id) throws Exception {
-        try {
-            hostDAO.deleteAllById(id);
-            hostAgentDAO.delete(id);
-            LOG.info("AgentJanitor delete all records for host {}.", id);
-        } catch (Exception e) {
-            LOG.error("AgentJanitor Failed to delete all records for host {}", id, e);
-        }
+        LOG.info(String.format("Delete records of stale host {}", id));
```

Comment:
Without this try catch the exception will be handled in 

SimpleAgentJanitor Run.
LOG.error("SimpleAgentJanitor Failed.", t);

Is this what we want?

Reply:
actually exception is caught in removeHost. but still better to catch uncaught. updated diff.